src/store/commit.js

```diff
@@ -0,0 +1,33 @@
+import { indexGenerator } from "../api/fetch";
+
+export function fetchAll({ commit, rootState }, { collection, commitAction }) {
```

Comment:
Passing the `index` function as a parameter would make this code more generic. Also, pass normal arguments as normal arguments. Creating an object to immediately destructure it doesn't make much sense.